src/runtime/Codecs/ListDecoder.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+
+namespace Python.Runtime.Codecs
+{
+    public class ListDecoder : IPyObjectDecoder
+    {
+        private static bool IsList(Type targetType)
+        {
+            if (!targetType.IsGenericType)
+                return false;
+
+            return targetType.GetGenericTypeDefinition() == typeof(IList<>);
+        }
+
+        private static bool IsList(PyObject objectType)
+        {
+            //must implement sequence protocol to fully implement list protocol
+            if (!SequenceDecoder.IsSequence(objectType)) return false;
+
+            //returns wheter the type is a list.
+            return objectType.Handle == Runtime.PyListType;

```

Comment:
Seems like this check is the only one needed.

Reply:
@lostmsu True, but eventually this should support any object that implements the list protocol (e.g. numpy arrays).